src/Symfony/Component/Routing/Loader/LoaderHelper.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Loader;
+
+/**
+ * Common parsing logic between route loaders.
+ *
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+trait LoaderHelper
+{
+    private function redirectToDefaults(string $redirect, bool $permanent = false): array
+    {
+        if ($this->looksLikeUrl($redirect)) {
+            $defaults['_controller'] = 'Symfony\Bundle\FrameworkBundle\Controller\RedirectController::urlRedirectAction';
+            $defaults['path'] = $redirect;
+        } else {
+            $defaults['_controller'] = 'Symfony\Bundle\FrameworkBundle\Controller\RedirectController::redirectAction';
+            $defaults['route'] = $redirect;
+        }
+
+        if ($permanent) {
+            $defaults['permanent'] = true;
+        }
+
+        return $defaults;
+    }
+
+    private function looksLikeUrl(string $urlOrRouteName): bool
```

Comment:
I fear any guessing strategy would be fragile. Instead, what about two defaults:
- `_redirect_to_route`
- `_redirect_to_uri`

Reply:
I don't think this is a valid fear: if the string starts with `/` or a `[scheme]://` then it's an URL seems enough to me. Having two ways of configuring the "target" reduces the DX.

Reply:
Moreover, we already do that guessing in other parts of the framework. E.g. the `login_path` option in the always delicate Security component.